### DISCUSSION TITLE
How to generate only TypeScript client interfaces and not classes?

### DISCUSSION THREAD

**Discussion by jcupps** [RicoSuter/NSwag#4818] [General]
Hello, is it possible to generate only the interfaces for the TypeScript clients and not the implementation classes? The `"typeStyle": "Interface"` setting only controls the DTOs. Setting `"generateClientInterfaces": true` does nothing when `generateClientClasses` is false. Thanks!

**Comment by sharky98** [comment]
I was able to generate only `interface` with the following command: `pnpm exec nswag openapi2tsclient /output:./interfaces.ts /input:./swagger.json /TypeScriptVersion:5.4 /GenerateClientClasses:false /TypeStyle:Interface /ImportRequiredTypes:true`. Looks like it is already what you have, so you may have something else not working.
